backends/bmv2/extractArchInfo.cpp

```diff
@@ -66,19 +65,19 @@ bool InferArchitecture::preorder(const IR::Type_Parser *node) {
 /// p4type = p4Type(param);
 /// if (p4type->is<IR::Type_Parser>) { visit(p4type); }
 /// else if (p4type->is<IR::Type_Control>) { visit(p4type); }
-bool InferArchitecture::preorder(const IR::Type_Package *node) {
+bool ExtractArchInfo::preorder(const IR::Type_Package *node) {
     for (auto p : node->getConstructorParameters()->parameters) {
         LOG1("package constructor param: " << p);
         auto ptype = typeMap->getType(p);
         if (ptype == nullptr)
             continue;
         if (ptype->is<IR::P4Parser>()) {
             LOG3("new parser model: " << p->toString());
-            v2model.parsers.push_back(new Parser_Model(p->toString()));
+            v2model.parsers.push_back(new P4::Parser_Model(p->toString()));
```

Comment:
It is dangerous to rely on the behavior of toString; maybe you can use instead p->name.